pages/client/edit_json/_clientId/index.vue

```diff
@@ -1,41 +1,86 @@
 <template>
   <div>
+    <b-modal
+      id="backupRestoreModal"
+      size="xl"
+      scrollable
+      centered
+      ok-title="Import"
+      :title="backupModalTitle"
+      @ok="onBackupRestore"
+    >
+      <JsonTree :data="backupJsonData" />
+    </b-modal>
     <b-row>
       <b-col>
         <h1>
-          Client {{ clientId }}: JSON Editor
-          <b-link class="float-right" @click="showJsonEditor = !showJsonEditor">
-            <b-icon-file-code v-if="showJsonEditor"></b-icon-file-code>
-            <b-icon-file-richtext v-else></b-icon-file-richtext>
-          </b-link>
+          {{ clientId }}
+          <span class="lead">Edit JSON configuration</span>
         </h1>
       </b-col>
     </b-row>
     <b-row>
       <b-col>
-        <b-button variant="danger" @click="$router.back()"> Cancel </b-button>
-      </b-col>
-      <b-col>
-        <b-button class="float-right" variant="primary" @click="saveConfig">
-          <b-spinner v-if="updating" class="ml-auto" small />
-          Save config
-        </b-button>
-      </b-col>
-    </b-row>
-    <b-row>
-      <b-col>
-        <JsonEditor
-          v-if="showJsonEditor"
-          v-model="jsonData"
-          :obj-data="jsonData"
-          is-edit="true"
-        ></JsonEditor>
-        <b-form-textarea
-          v-else
-          :value="jsonData"
-          max-rows="30"
-          @input="jsonData = JSON.parse($event)"
-        ></b-form-textarea>
+        <b-card no-body>
+          <b-tabs card>
+            <b-tab title="Graphical Editor" :active="showJsonEditor" lazy>
+              <b-container>
+                <JsonEditor
+                  v-model="jsonData"
+                  :obj-data="jsonData"
+                  is-edit="true"
+                />
+              </b-container>
+            </b-tab>
+            <b-tab title="Textual Editor" :active="!showJsonEditor" lazy>
+              <b-form-textarea
+                ref="rawInput"
+                :value="jsonData"
+                max-rows="25"
+                class="rawInput"
+                autocorrect="off"
+                spellcheck="false"
+                data-gramm="false"
+                @input="onRawEdit($event)"
+              />
+            </b-tab>
+            <b-tab title="Preview" lazy>
+              <json-tree :data="jsonData" />
+            </b-tab>
+            <b-tab title="Backups">
+              <b-list-group>
+                <b-list-group-item
+                  v-for="backup in backups"
+                  :key="backup"
+                  button
+                  @click="onBackupClick(backup)"
```

Comment:
```suggestion
                  v-for="backupId in backupIds"
                  :key="backupId"
                  button
                  @click="onBackupClick(backupId)"
```

Consistency with other code and it's clear that we're not handling complex objects here.